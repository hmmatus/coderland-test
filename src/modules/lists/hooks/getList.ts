import { useQuery } from '@tanstack/react-query';
export function getListHook() {
  return useQuery({
    queryKey: ['list'],
    queryFn: async () => {
      const response = await fetch('https://6172cfe5110a740017222e2b.mockapi.io/elements');
      return response.json();
    },
  });
}
