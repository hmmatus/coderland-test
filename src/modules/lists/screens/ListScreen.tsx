import React from 'react';
import { ActivityIndicator, FlatList, StyleSheet, Text, View } from 'react-native';
import ScreenWrapper from '../../core/components/wrappers/ScreenWrapper';
import { useQuery } from '@tanstack/react-query';
import ListElementCard from '../components/cards/ListElementCard';
import { getListHook } from '../hooks/getList';
import LoadingLayout from '../../core/components/layouts/LoadingLayout';
const ListScreen = () => {
  const { data, isLoading, isError } = getListHook();
  if (isLoading) return <LoadingLayout />;
  return (
    <ScreenWrapper>
      <View style={styles.container}>
        <FlatList
          data={data}
          keyExtractor={(item) => item.id}
          ItemSeparatorComponent={() => <View style={{ height: 12 }} />}
          renderItem={({ item }) => <ListElementCard item={item} />}
        />
      </View>
    </ScreenWrapper>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    padding: 12,
  },
});

export default ListScreen;
