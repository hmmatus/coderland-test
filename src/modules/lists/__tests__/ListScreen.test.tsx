import { render, renderHook, screen, waitFor } from '@testing-library/react-native';
import React from 'react';
import ListScreen from '../screens/ListScreen';
import { QueryClient, QueryClientProvider, useQuery } from '@tanstack/react-query';
import { getListHook } from '../hooks/getList';
const queryClient = new QueryClient();
const wrapper = ({ children }: { children: JSX.Element }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);
describe('Testing List Page', () => {
  it('Should render properly', () => {
    const newQueryClient = new QueryClient();
    render(
      <QueryClientProvider client={newQueryClient}>
        <ListScreen />
      </QueryClientProvider>
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
  it('Should render the list of elements', async () => {
    const { result } = renderHook(() => getListHook(), { wrapper });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));
  });
});
