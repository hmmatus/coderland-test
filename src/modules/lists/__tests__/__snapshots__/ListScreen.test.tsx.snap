// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List Page Should render properly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    color="blue"
    size="large"
  />
  <Text>
    Loading...
  </Text>
</RCTSafeAreaView>
`;
